resources/lib/navigation/directory_search.py

```diff
@@ -155,7 +153,9 @@ def search_clear():
     if not ui.ask_for_confirmation(common.get_local_string(30404), common.get_local_string(30406)):
         return False
     G.LOCAL_DB.clear_search_items()
-    search_list(dir_update_listing=True)
+    search_list(dir_update_listing=False)
+    url = common.build_url(['search', 'search'], mode=G.MODE_DIRECTORY)
+    common.container_update(url, True)
```

Comment:
i do not understand the use case of this change

Reply:
- Clear search results
- Add search
- Press back
- Dialog asking to clear results re-triggers because path in history is `/search/search/clear/`

Using Container.Update(path,replace) will rewrite the path history so that `/search/search/clear/` is replaced with `/search/search/` and thus pressing back from search results will no longer re-trigger the clear dialog.

Reply:
Other option is to use the same method as I used in the other function as shown here:
https://github.com/jurialmunkey/plugin.video.netflix/commit/5b0f8835cb418ea7a4c4362d54457ae2f64baee8

This approach is slightly better because it keeps the path history before the clear. I'm just not familiar enough with the plugin code to be certain whether passing through `dir_update_listing` to `search_list()` in the `route_search_nav` function would inadvertently cause other issues - so I just stuck with what I knew was definitely safe.

Reply:
if you clear the path history `common.container_update(url, True)`
you can no longer return to the main menu after cleared the list

Reply:
i do not know if this situation can be full resolved in easy way,
kodi does not provide many ways to handle these particular situations
as a last resort the Clear history menu could be moved as context menu

Reply:
If you're concerned about returning to the main menu then see the commit that I linked above. That also resolves the issue and does not destroy history.

This one https://github.com/jurialmunkey/plugin.video.netflix/commit/5b0f8835cb418ea7a4c4362d54457ae2f64baee8

Reply:
i see but this broken the history when you perform the use case that you explained in second post
i think because the param container_update is recursive

Reply:
No it doesn't break the history and the param is not recursive. Did you actually try the commit I linked?

Here's a video demonstrating that it retains history with exactly that use case.
https://www.youtube.com/watch?v=75svKp0s0xQ

Reply:
yes works for the use case of 2° post,
but if you delete only the history and go back to main menu the back button not work well
you have to press back two times

Reply:
See latest commit I pushed. Should solve that issue.

Clear method listitem now has `is_folder` to False and uses `Container.Refresh()` to update container instead. The `media_type` key is set to blank string to skip setting `isPlayable` property flag in `_convert_dict_to_listitem` (and also to avoid missing key error).